### PR TITLE
Add endsWith() and startsWith() methods to Bart\Primitives\Strings.

### PR DESCRIPTION
**Pull request by nadeemahmad**
I am not sure whether to leave the stackoverflow reference or not.

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @nadeemahmad, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: test/Bart/Primitives/StringsTest.php [hunk 1]
```diff
@@ -43,6 +43,48 @@ public function dataProviderTestSummarizeWithSuffix()
 		];
 	}
 
+    public function dataProviderTestStartsWith()
+    {
+        return [
```
Comment (asheepapart) [inline test/Bart/Primitives/StringsTest.php]: Try one where the length of substring is greater than full string when full string is not null. 
Also, one in which they are both null. 
Also try some invalid inputs like numbers.

Comment (nadeemahmad) [inline test/Bart/Primitives/StringsTest.php]: Done.

**Comment by nadeemahmad** [comment]
In the latest commit, I added support for actual string validation. It will now ensure that the passed in argument is indeed a string, and throw an exception if it is not. I feel like this is more comprehensive.

**Comment by nadeemahmad** [comment]
Is this good to merge?

**Comment by asheepapart** [comment]
I'd advise changing the validate method to take only a single arg to
validate.

**Comment by nadeemahmad** [comment]
Yeah that makes more sense.

**Pull request merged**